components/lsp-form/MintForm.tsx

```diff
@@ -9,19 +9,103 @@ import {
 } from "./LSPForm.styled";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faArrowDown } from "@fortawesome/free-solid-svg-icons";
+import { ethers } from "ethers";
+import toWeiSafe from "../../utils/convertToWeiSafely";
 
 import LongShort from "./LongShort";
 import Collateral from "./Collateral";
 
+const toBN = ethers.BigNumber.from;
+const scaledToWei = toBN("10").pow("18");
+
+// max uint value is 2^256 - 1
+const MAX_UINT_VAL = ethers.constants.MaxUint256;
+const INFINITE_APPROVAL_AMOUNT = MAX_UINT_VAL;
+
 interface Props {
+  address: string;
+  contractAddress: string;
   setShowSettle: React.Dispatch<React.SetStateAction<boolean>>;
+  lspContract: ethers.Contract | null;
+  erc20Contract: ethers.Contract | null;
+  web3Provider: ethers.providers.Web3Provider | null;
+  collateralBalance: ethers.BigNumber;
+  collateralPerPair: ethers.BigNumber;
+  setCollateralBalance: React.Dispatch<React.SetStateAction<ethers.BigNumber>>;
+  collateralDecimals: string;
+  longTokenBalance: ethers.BigNumber;
+  longTokenDecimals: string;
+  shortTokenBalance: ethers.BigNumber;
+  shortTokenDecimals: string;
+  refetchLongTokenBalance: () => void;
+  refetchShortTokenBalance: () => void;
 }
-const MintForm: FC<Props> = ({ setShowSettle }) => {
-  const [collateral, setCollateral] = useState("");
+
+const MintForm: FC<Props> = ({
+  lspContract,
+  erc20Contract,
+  contractAddress,
+  collateralBalance,
+  collateralPerPair,
+  address,
+  setCollateralBalance,
+  collateralDecimals,
+  longTokenBalance,
+  longTokenDecimals,
+  shortTokenBalance,
+  shortTokenDecimals,
+  refetchLongTokenBalance,
+  refetchShortTokenBalance,
+}) => {
+  const [collateral, setCollateral] = useState("uma");
   const [amount, setAmount] = useState("");
   const [longTokenAmount, setLongTokenAmount] = useState("");
   const [shortTokenAmount, setShortTokenAmount] = useState("");
 
+  const mint = useCallback(async () => {
+    if (lspContract && erc20Contract && amount) {
+      const weiAmount = toWeiSafe(amount);
+      try {
+        const allowance = await erc20Contract.allowance(
+          address,
+          contractAddress
+        );
+        const balance = await erc20Contract.balanceOf(address);
+        const hasToApprove = allowance.lt(balance);
+        if (hasToApprove) {
+          const approveTx = await erc20Contract.approve(
+            contractAddress,
+            INFINITE_APPROVAL_AMOUNT
+          );
+
+          await approveTx.wait();
+        }
+        // Need to send the correct amount based on the collateral pair ** the amount
+        // User has specified in the input.
+        const ratio = scaledToWei.div(collateralPerPair);
+
+        lspContract
+          .create(weiAmount.mul(ratio))
```

Comment:
I think this will break if collateralPerPair is < 1e18 (aka less than one unit of collateral per pair of tokens). In that case, the ratio will round to 0 and you will mint 0 tokens no matter what the user inputs. What you want is to do _all_ the multiplication first and then _all_ the division at the end to minimize the rounding. Note: this will mean you will not be able to have well-reasoned intermediate values, like ratio, so the code will be a little less understandable :/.

```ts
// All operations that make the number larger come first. All the operations that make the number smaller come last.
const mintAmount = weiAmount.mul(scaledtoWei).div(collateralPerPair);
lspContract.create(mintAmount).then(...);
```